### ISSUE TITLE
Custom dataset classes not detected

### ISSUE THREAD

**Issue by mursabogdan1** [yassouali/pytorch_segmentation#72]
First of all, I would like to express my appreciation for the work you spent on this project - a fantastic toolkit that helps a lot the productivity for image segmentation tasks.

I am writing this issue not to raise a problem regarding the project, but more about a question I am dealing with.

I want to train one of the models implemented using a custom dataset with five classes (including background); hence I created my own data loader using ade20k as a template.

I did not do too many things, but a few adjustments on the config's hyperparameters and initiated a training - but 2 of the classes are not detected at all. I did check both training/testing dataset for an eventual unbalanced dataset but no problem there.

Do you have any idea about what could be the cause of the problem or where I should start at for a more in-depth investigation?

I tried to follow all indication provided in your readme, such as setting:
```bash
   ignore_index = -1
```

or 

```python
self.num_classes = 5
```

or adjust self.MEAN, self.STD accordingly with the dataset images.

In the end, no matter the number of epochs passed, the last 2 classes keep indicating a 0 mIoU.

```python
        ## Info for epoch 5 ## 
         val_loss       : 0.34638
         Pixel_Accuracy : 0.84
         Mean_IoU       : 0.375
         Class_IoU      : {0: 0.83, 1: 0.458, 2: 0.586, 3: 0.0, 4: 0.0}
```

Still, adding an "extra" class hence 6 classes, I get the following output:
```python
     ## Info for epoch 15 ## 
         val_loss       : 0.30863
         Pixel_Accuracy : 0.88
         Mean_IoU       : 0.34599998593330383
         Class_IoU      : {0: 0.879, 1: 0.514, 2: 0.671, 3: 0.0, 4: 0.0, 5: 0.009}
```

Somehow 3 and 4 indexes are not actually my classes, and 5 seems to be one of them .. and probably the last one finds itself somewhere between 5 and 149. 

Am I missing something?

Thank you in advance.

**Comment by mursabogdan1** [comment]
Interestingly, while performing a 
```python
label = np.asarray(Image.open(label_path), dtype=np.int32) - 1
np.unique(label)
```

I get as output:
```python
array([-1,  0,  1,  4,  5], dtype=int32)
```

which means that my indexes are ordered labels. 

It seems to be a rookie mistake, but should I perform any greyscaling of the pictures before?

**Comment by mursabogdan1** [comment]
Jinx - involuntary I did exactly how you recommended.  Thank you very much!

Overall, do you consider that it is better to have a background as a label instead of ignoring it?

**Comment by yassouali** [comment]
If the background is correctly annotated, yes, better to use it I guess.
